packages/yoastseo/src/values/Mark.js

```diff
@@ -108,6 +109,36 @@ Mark.prototype.serialize = function() {
 	};
 };
 
+/**
+ * Checks if the mark object is valid.
+ * @returns {void}
+ */
+// eslint-disable-next-line complexity
+Mark.prototype.isValid = function() {
+	if ( ! isUndefined( this.getPositionStart() ) && this.getPositionStart() < 0 ) {
+		throw new RangeError( "positionStart should be larger or equal than 0." );
+	}
+	if ( ! isUndefined( this.getPositionEnd() ) && this.getPositionEnd() <= 0 ) {
+		throw new RangeError( "positionEnd should be larger than 0." );
+	}
+	if ( ! isUndefined( this.getPositionStart() ) && ! isUndefined( this.getPositionEnd() ) &&
+		this.getPositionStart() >= this.getPositionEnd() ) {
+		throw new RangeError( "The positionStart should be smaller than the positionEnd." );
+	}
+	if ( isUndefined( this.getPositionStart() ) && ! isUndefined( this.getPositionEnd() ) ||
+		 isUndefined( this.getPositionEnd() ) && ! isUndefined( this.getPositionStart() ) ) {
+		throw new Error( "A mark object should either have start and end defined or start and end undefined." );
+	}
+};
+
+/**
+ * Checks if a mark has position information available.
+ * @returns {boolean} Returns true if the Mark object has position information, false otherwise.
+ */
+Mark.prototype.hasPosition = function() {
+	return this.getPositionStart && this.getPositionStart();
```

Comment:
If I understand this line correctly, we check whether `getPositionStart` is defined and execute that method if it is. However, that method returns a number. So wouldn't it also make `hasPosition()` returns a number not a boolean? 🤔

Reply:
Adapted!